astropy/table/tests/test_index.py

```diff
@@ -460,3 +477,34 @@ def test_copy_indexed_table(self, table_types):
             for index, indexp in zip(t.indices, tp.indices):
                 assert np.all(index.data.data == indexp.data.data)
                 assert index.data.data.colnames == indexp.data.data.colnames
+
+    def test_updating_row_byindex(self, main_col, table_types, engine):
+        self._setup(main_col, table_types)
+        t = Table([[1, 2, 3, 4], [2, 3, 4, 5], [3, 4, 5, 6]], names=('a', 'b', 'c'), meta={'name': 'first table'})
+
+        t.add_index('a', engine=engine)
+        t.add_index('b', engine=engine)
+
+        t.loc[3] = [3,4,5] # single label, with primary key 'a'
```

Comment:
These values are the same as the original, so not a good test.  Make them something very different like [30, 40, 50].  This brings up a question about whether the `a` and `b` indexes will get updated correctly when you change index key values like this.  I'm not sure if this will or should work...

Reply:
just now tested with your value and its working fine.will update it!

Reply:
shall i change the first column to strings here?